`test_roundtrip` failing builds

https://github.com/xpublish-community/xpublish/blob/c4134dbc074ec1fa5a2cc7e55f458b5f98a6a34e/tests/test_zarr_compat.py#L140-L156

Is causing [builds](https://github.com/xpublish-community/xpublish/actions/runs/6499784421/job/17653846996?pr=236#step:6:6296) with unrelated changes to fail recently.

It looks like the issue is due to the time encoding.
```py
FAILED tests/test_zarr_compat.py::test_roundtrip[2018-01-01-2021-01-01-6H-180-360-True-gregorian-False] - AssertionError: Left and right Dataset objects are not identical

Differing coordinates:
L * time         (time) datetime64[ns] 2018-01-01 2018-01-01 ... 2020-12-31
    bounds: time_bounds
R * time         (time) datetime64[ns] 2018-01-01 ... 2020-12-31T12:00:00
    bounds: time_bounds
Differing data variables:
L   tmin         (time, lat, lon) float32 dask.array<chunksize=(4384, 180, 360), meta=np.ndarray>
R   tmin         (time, lat, lon) float32 dask.array<chunksize=(4384, 180, 360), meta=np.ndarray>
L   time_bounds  (time, d2) datetime64[ns] dask.array<chunksize=(4384, 2), meta=np.ndarray>
R   time_bounds  (time, d2) datetime64[ns] dask.array<chunksize=(4384, 2), meta=np.ndarray>
L   tmax         (time, lat, lon) float32 dask.array<chunksize=(4384, 180, 360), meta=np.ndarray>
R   tmax         (time, lat, lon) float32 dask.array<chunksize=(4384, 180, 360), meta=np.ndarray>
```

A warning is getting thrown earlier
```py
  /home/runner/work/xpublish/xpublish/xpublish/utils/zarr.py:144: UserWarning: Times can't be serialized faithfully to int64 with requested units 'days since 1980-01-01'. Resolution of 'hours' needed. Serializing times to floating point instead. Set encoding['dtype'] to integer dtype to serialize to int64. Set encoding['dtype'] to floating point dtype to silence this warning.
    encoded_da = encode_zarr_variable(dvar, name=key)
```

Which is from here:

https://github.com/xpublish-community/xpublish/blob/c4134dbc074ec1fa5a2cc7e55f458b5f98a6a34e/xpublish/utils/zarr.py#L132-L155

But is really coming from the Xarray Zarr backend.

https://github.com/pydata/xarray/blob/25e6e084aa18c49d92934db298a3efff9c712766/xarray/backends/zarr.py#L288-L318

--- a/dev-requirements.txt
+++ b/dev-requirements.txt
@@ -16,5 +16,5 @@ pytest-cov
 requests
 toolz
 uvicorn
-xarray
+xarray!=v2023.09.0  # see https://github.com/xpublish-community/xpublish/issues/237
 zarr

--- a/requirements.txt
+++ b/requirements.txt
@@ -6,5 +6,5 @@ numpy
 pluggy
 toolz
 uvicorn
-xarray
+xarray!=v2023.09.0  # see https://github.com/xpublish-community/xpublish/issues/237
 zarr

